Subject: [PATCH] 5.21 Сортировка выборкой
---
Index: src/main/java/ru/array/SortSelected.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/array/SortSelected.java b/src/main/java/ru/array/SortSelected.java
--- a/src/main/java/ru/array/SortSelected.java	(revision 8755e96f3545e8bf2a524053dc6cdea7ffab90f6)
+++ b/src/main/java/ru/array/SortSelected.java	(revision b21988331d3351b9dd998d3a51d7ef3caf9b8bbe)
@@ -3,7 +3,7 @@
 public class SortSelected {
 
     public static int[] sort(int[] data) {
-        for (int i = 0; i < data.length; i++) {
+        for (int i = 0; i < data.length - 1; i++) {
             int min = MinDiapason.findMin(data, i, data.length - 1);
             int index = FindLoop.indexInRange(data, min, i, data.length - 1);
             SwitchArray.swap(data, i, index);
Index: src/test/java/ru/array/SortSelectedTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/ru/array/SortSelectedTest.java b/src/test/java/ru/array/SortSelectedTest.java
--- a/src/test/java/ru/array/SortSelectedTest.java	(revision 8755e96f3545e8bf2a524053dc6cdea7ffab90f6)
+++ b/src/test/java/ru/array/SortSelectedTest.java	(revision b21988331d3351b9dd998d3a51d7ef3caf9b8bbe)
@@ -22,7 +22,7 @@
 
     @Test
     public void whenSort3() {
-        int[] data = new int[] {2, 3, 1};
+        int[] data = new int[] {2, 6, 1};
         int[] result = SortSelected.sort(data);
         int[] expected = new int[] {1, 2, 3};
         assertThat(result).containsExactly(expected);
